dataset/dataset.cpp

```diff
@@ -232,20 +225,28 @@ void union_op_in_place(Masks &masks, const Masks &otherMasks, Op op) {
   for (const auto &[key, item] : otherMasks) {
     const auto it = masks.find(key);
     if (it != masks.end()) {
-      op(it->second, item);
+      if (it->second.is_readonly()) {
+        if (it->second != op(copy(it->second), item))
+          throw except::DimensionError(
+              "Cannot update mask via slice since the mask is being broadcast "
+              "along the slice dimension.");
```

Comment:
The error message does not sound right. It looks like the if checks for any change including the values and not just the shape.